cmd/kubelet/app/options/options.go

```diff
@@ -544,4 +544,6 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 	fs.StringSliceVar(&c.EnforceNodeAllocatable, "enforce-node-allocatable", c.EnforceNodeAllocatable, "A comma separated list of levels of node allocatable enforcement to be enforced by kubelet. Acceptable options are 'none', 'pods', 'system-reserved', and 'kube-reserved'. If the latter two options are specified, '--system-reserved-cgroup' and '--kube-reserved-cgroup' must also be set, respectively. If 'none' is specified, no additional options should be set. See https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/ for more details.")
 	fs.StringVar(&c.SystemReservedCgroup, "system-reserved-cgroup", c.SystemReservedCgroup, "Absolute name of the top level cgroup that is used to manage non-kubernetes components for which compute resources were reserved via '--system-reserved' flag. Ex. '/system-reserved'. [default='']")
 	fs.StringVar(&c.KubeReservedCgroup, "kube-reserved-cgroup", c.KubeReservedCgroup, "Absolute name of the top level cgroup that is used to manage kubernetes components for which compute resources were reserved via '--kube-reserved' flag. Ex. '/kube-reserved'. [default='']")
+	fs.StringVar(&c.LoggingConfig.LoggingFormat, "logging-format", c.LoggingConfig.LoggingFormat, `Sets the log format. Permitted formats: "text", "json".\nNon-default formats don't honor these flags: -add_dir_header, --alsologtostderr, --log_backtrace_at, --log_dir, --log_file, --log_file_max_size, --logtostderr, --skip_headers, --skip_log_headers, --stderrthreshold, --log-flush-frequency.\nNon-default choices are currently alpha and subject to change without warning.`)
```

Comment:
so if I use json logging, I can't direct log output to a file or rotate logs? that's really surprising to me. can we at least catch use of unsupported combinations of log options to prevent confusion?

Reply:
We had similar discussion with @lavalamp here https://github.com/kubernetes/kubernetes/pull/92177#discussion_r442928972
We are planning to prevent from use of unsupported combinations, it's already being worked on by @rahulchheda.
Still we don't want to block effort on implementing this, that's way we explicitly list those flags and mention that non default config is in ALPHA.

Reply:
We'll wait for the protecting PR to be ready before merging this. We avoid merging PRs that expand API surface (in this case, CLI flags) ahead of protections that make sure the API isn't accidentally used incorrectly.

Reply:
Do we want to protect users even from features explicitly maked as ALPHA?

Reply:
if it breaks existing non-alpha features like log file and log rotation, yes